### DISCUSSION TITLE
Next13: useColorModeValue fires twice for both light and dark modes

### DISCUSSION THREAD

**Discussion by saiy2k** [chakra-ui/chakra-ui#8920] [General]
### Description

The following code

      <Image
        src={useColorModeValue(Logo, LogoWhite)}
        alt="logo"
        width="371"
        height="68"
      />

should load only the white/dark logo, based on the current theme.

I have also set config.initialColorMode in layout.tsx as follows:
```
let themeValue;
if (typeof window !== "undefined")
  themeValue = localStorage.getItem("chakra-ui-color-mode") || "dark";
else themeValue = "dark";

const theme = extendTheme({
  colors,
  config: {
    initialColorMode: themeValue
  }
});
```

Since I am passing "dark" as default value, in dark theme, it works fine. But in light theme, it loads both the images and image flashes.

Tried turning off `ssr` for layout.tsx, but getting this error: `invariant expected app router to be mounted`

Created a 'sub' folder and tried doing `ssr: off` in there and it works there. Verified even in network tab, that only the theme respective image loads on page load.

Need to make this work with the root layout.

Any inputs will be highly appreciated. Thanks.

Links
Repo to reproduce this issue: https://github.com/saiy2k/Nextjs-Chakra-Dark-theme-PoC

Vercel deployment for quick preview:
https://nextjs-chakra-dark-theme-po-c.vercel.app/ --> Page with issue (Can't turn off ssr)
https://nextjs-chakra-dark-theme-po-c.vercel.app/sub --> Page working as expected. With ssr: off, only the theme specific image is loaded.

Codesandbox: https://codesandbox.io/p/sandbox/issue-chakra-ui-next13-usecolormodevalue-tsdyvh

### Link to Reproduction

https://codesandbox.io/p/sandbox/issue-chakra-ui-next13-usecolormodevalue-tsdyvh

### Steps to reproduce

1. Open 'Network tab' in Chrome and set filter to 'Img'
2. Navigate to https://nextjs-chakra-dark-theme-po-c.vercel.app/ (If dark theme is loaded by default, switch to the white theme and reload)
3. See both 'logo-white.png' and 'logo.png' are loaded.
4. Navigate to https://nextjs-chakra-dark-theme-po-c.vercel.app/sub
5. See only 'logo.png' is loaded.

### Chakra UI Version

2.5.5

### Operating System

- [ ] macOS
- [ ] Windows
- [X] Linux

**Comment by godfrednanaowusu** [comment]
same thing happening here

**Comment by BEDIRHANSAMSA** [comment]
same here

**Comment by Jpac14** [comment]
Same here
